torch/distributions/utils.py

```diff
@@ -130,25 +131,36 @@ def probs_to_logits(probs, is_binary=False):
     return torch.log(ps_clamped)
 
 
-class lazy_property:
+class lazy_property(property):
     r"""
     Used as a decorator for lazy loading of class attributes. This uses a
     non-data descriptor that calls the wrapped method to compute the property on
     first call; thereafter replacing the wrapped method into an instance
     attribute.
     """
 
-    def __init__(self, wrapped):
-        self.wrapped = wrapped
-        update_wrapper(self, wrapped)  # type:ignore[arg-type]
-
-    def __get__(self, instance, obj_type=None):
-        if instance is None:
-            return _lazy_property_and_property(self.wrapped)
-        with torch.enable_grad():
-            value = self.wrapped(instance)
-        setattr(instance, self.wrapped.__name__, value)
-        return value
+    def __init__(self, fget=None, fset=None, fdel=None, doc=None):
+        wrapped = fget
+        assert fset is None
+        assert doc is None
+        wrapped = wrapped
+        name = "_" + wrapped.__name__
```

Comment:
It would be safer and slightly clearer to set `name = "_lazy_" + wrapped.__name__`.  I just worry that among the thousands of user-defined distributions in the wild, many already have members with names like `._logits` and `._scale` that would conflict with a mere "_" prefix, and I'd bet there will be zero conflicts in the wild with a "_lazy_" prefix.